### DISCUSSION TITLE
How do I list installed packages when using manifest mode?

### DISCUSSION THREAD

**Discussion by sketch34** [microsoft/vcpkg#20523] [Q&A] [answered]
I just switched to manifest mode, everything builds fine. However passing `list` to the CLI now returns nothing:

```
$ ./submodules/vcpkg/vcpkg list
No packages are installed. Did you mean `search`?
```

Before switching to manifest mode this would list all my installed packages.

**Comment by dg0yt** [comment, marked as answer]
If you are doing this in the source directory of an out-of-source build: vcpkg doesn't know your actual target configuration, with an installed tree in the binary directory.
But perhaps you can try a `vcpkg install --dry-run` to see what would be installed for the default triplet?

**Reply by sketch34** [reply]
Thank you that works. Perhaps the the `list` command could detect that manifest mode is enabled (via the presence of `vcpkg.json`)?

**Reply by dg0yt** [reply]
The `vpckg list` command works with the same installed tree as a plain `vcpkg install` in manifest mode. Unfortunately this is not the same root that is used when `vcpkg` is invoked by `cmake`...

**Reply by sketch34** [reply]
Could the install trees be made the same by changing the vcpkg cmake toolchain file? From the docs it seems like the intention is that manifest mode vs CLI mode should function exactly the same from the user's perspective.

My naive mental model was that `vcpkg.json` is basically just automating the CLI.
